tst/EnergyPlus/unit/TestHelpers/IOHelper.cc

```diff
@@ -0,0 +1,508 @@
+#include <EnergyPlus/DataIPShortCuts.hh>
+
+#include <fstream>
+#include <algorithm>
+
+struct InputProcessorCache;
+std::unique_ptr<EnergyPlus::InputProcessorCache> EnergyPlus::IOHelper::m_idd_cache = nullptr;
+
+namespace EnergyPlus {
+
+	IOHelper::IOHelper()
+	{
+		this->eso_stream = std::unique_ptr< std::ostringstream >( new std::ostringstream );
+		this->mtr_stream = std::unique_ptr< std::ostringstream >( new std::ostringstream );
+		this->echo_stream = std::unique_ptr< std::ostringstream >( new std::ostringstream );
+
+		DataGlobals::eso_stream = this->eso_stream.get();
+		DataGlobals::mtr_stream = this->mtr_stream.get();
+		InputProcessor::echo_stream = this->echo_stream.get();
+
+		m_cout_buffer = std::unique_ptr< std::ostringstream >( new std::ostringstream );
+		m_redirect_cout = std::unique_ptr< RedirectCout >( new RedirectCout( m_cout_buffer ) );
+
+		m_cerr_buffer = std::unique_ptr< std::ostringstream >( new std::ostringstream );
+		m_redirect_cerr = std::unique_ptr< RedirectCerr >( new RedirectCerr( m_cerr_buffer ) );
+	}
+
+	IOHelper::~IOHelper()
+	{
+		DataGlobals::eso_stream = nullptr;
+		DataGlobals::mtr_stream = nullptr;
+		InputProcessor::echo_stream = nullptr;
+	}
+
+	void IOHelper::setup_cache()
+	{
+		if ( ! m_idd_cache ) {
+			static auto errors_found = false;
+			static auto const idd = "";
+			InputProcessor::clear_state();
+			process_idd( idd, errors_found );
+			if ( errors_found ) {
+				InputProcessor::clear_state();
+				return;
+			}
+			m_idd_cache = std::unique_ptr< InputProcessorCache >( new InputProcessorCache );
+			InputProcessor::clear_state();
+		}
+	}
+
+	void IOHelper::use_cache()
+	{
+		setup_cache();
+		if ( m_idd_cache ) {
+			m_idd_cache->use_cached_namespace_variables();
+		}
+	}
+
+	template < typename T, class T2 >
+	bool IOHelper::compare_containers( T const & expected_container, T2 const & actual_container ) {
+		bool is_valid = ( expected_container.size() == actual_container.size() );
+		EXPECT_EQ( expected_container.size(), actual_container.size() ) << "Containers are not equal size.";
+		auto expected = expected_container.begin();
+		auto actual = actual_container.begin();
+		for ( ; expected != expected_container.end(); ++expected, ++actual ) {
+			// This may fail due to floating point issues for float and double...
+			EXPECT_EQ( *expected, *actual ) << "Incorrect index: " << ( expected - expected_container.begin() );
+			is_valid = ( *expected == *actual );
+		}
+		return is_valid;
+	}
+
+	std::string IOHelper::delimited_string( std::vector<std::string> const & strings, std::string const & delimiter ) {
+		std::ostringstream compare_text;
+		for( auto const & str : strings ) {
+			compare_text << str << delimiter;
+		}
+		return compare_text.str();
+	}
+
+	bool IOHelper::compare_eso_stream( std::string const & expected_string, bool reset_stream ) {
+		auto const stream_str = this->eso_stream->str();
+		EXPECT_EQ( expected_string, stream_str );
```

Comment:
This is surprising to me.  An assertion down in a supporting function.  Then you return the result of comparison.  Am I as a client of this thing supposed to put an assertion around that return value, thus doing the job twice?  Or I am I supposed to know you hide an assertion in your implementation?